crates/core/src/checkpoint.rs

```diff
@@ -40,6 +40,8 @@ bucket_list(bucket, checksum) AS (
         json_extract(json_each.value, '$.bucket') as bucket,
         json_extract(json_each.value, '$.checksum') as checksum
   FROM json_each(json_extract(?1, '$.buckets'))
+  WHERE IFNULL(json_extract(json_each.value, '$.priority'), 1) <=
+    IFNULL(json_extract(?1, '$.priority'), 3)
```

Comment:
What is this check used for?

Reply:
The idea is that clients would send both the target priority and the priority of each bucket. Then we'd only validate the checksums of buckets matching that priority.

Clients can of course just send the buckets with a matching priority to begin with, which makes this check superfluous.